upcloud/request/server_group.go

```diff
@@ -91,11 +117,23 @@ func (s ModifyServerGroupRequest) RequestURL() string {
 
 // MarshalJSON is a custom marshaller that deals with deeply embedded values.
 func (r ModifyServerGroupRequest) MarshalJSON() ([]byte, error) {
-	type c ModifyServerGroupRequest
 	v := struct {
-		ServerGroup c `json:"server_group"`
+		ServerGroup serverGroupPayload `json:"server_group"`
 	}{}
-	v.ServerGroup = c(r)
+
+	v.ServerGroup = serverGroupPayload{
+		Labels:  r.Labels,
+		Members: r.Members,
+		Title:   r.Title,
+	}
+
+	if !r.AntiAffinity.Empty() {
```

Comment:
This logic is (partially) duplicated, because the payload for modify and create requests is actually slightly different - in create `Members` is a slice, in modify `Members` is a pointer to a slice